ipatests/test_integration/test_replica_promotion.py

```diff
@@ -477,6 +477,30 @@ def test_manual_renewal_master_transfer(self):
             "Replica hostname not found among CA renewal masters"
         )
 
+    def test_renewal_master_with_csreplica_manage(self):
+
+        master = self.master
+        replica = self.replicas[0]
+
+        master.run_command(['ipa-csreplica-manage', 'set-renewal-master',
+                            '-p', master.config.dirman_password])
+        result = master.run_command(["ipa", "config-show"]).stdout_text
+
+        assert("IPA CA renewal master: %s" % master.hostname in result), (
```

Comment:
The new test case implements the ticket. You could go one tiny step further and also validate that there is only one renewal master after running ``set-renewal-master``. It would be a bug to have two renewal masters or have different settings on master and replica.

```
def assertCARenewalMaster(self, host, expected):
    result = host.run_command(["ipa", "config-show"]).stdout_text
    matches = list(regex.finditer('IPA CA renewal master: (.*)', result))
    assert len(matches), 1
    assert matches[0].group(1) == expected

def test...(self)
    ...
    self.assertCARenewalMaster(master, master.hostname)
    self.assertCARenewalMaster(replica, master.hostname)
    # switch renewal master
    self.assertCARenewalMaster(master, replica.hostname)
    self.assertCARenewalMaster(replica, replica.hostname)
```